packages/blade/src/components/Table/docs/APIStories/VirtualizedTableAPI.stories.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import type { StoryFn, Meta } from '@storybook/react';
+import type { TableData } from '../../types';
+import { Table as TableComponent } from '../../Table';
+import { TableHeader, TableHeaderRow, TableHeaderCell } from '../../TableHeader';
+import { TableRow, TableCell, VirtulizedWrapper } from '../../TableBody';
+import StoryPageWrapper from '~utils/storybook/StoryPageWrapper';
+import { Box } from '~components/Box';
+import { Amount } from '~components/Amount';
+import { Code } from '~components/Typography';
+import { Badge } from '~components/Badge';
+
+export default {
+  title: 'Components/Table/API',
+  component: VirtulizedWrapper,
+  args: {},
+  argTypes: {
+    children: {
+      control: {
+        disable: true,
+      },
+    },
+  },
+  parameters: {
+    docs: {
+      page: () => (
+        <StoryPageWrapper
+          componentDescription="You can find a complete list of TableVirtulized props here"
+          componentName="TableVirtulized"
+          apiDecisionComponentName="Table"
+        />
+      ),
+    },
+  },
+} as Meta<typeof VirtulizedWrapper>;
+
+const nodes: Item[] = [
+  ...Array.from({ length: 5000 }, (_, i) => ({
+    id: (i + 1).toString(),
+    paymentId: `rzp${Math.floor(Math.random() * 1000000)}`,
+    amount: Number((Math.random() * 10000).toFixed(2)),
+    status: ['Completed', 'Pending', 'Failed'][Math.floor(Math.random() * 3)],
+    date: new Date(2021, Math.floor(Math.random() * 12), Math.floor(Math.random() * 28) + 1),
+    type: ['Payout', 'Refund'][Math.floor(Math.random() * 2)],
+    method: ['Bank Transfer', 'Credit Card', 'PayPal'][Math.floor(Math.random() * 3)],
+    bank: ['HDFC', 'ICICI', 'SBI'][Math.floor(Math.random() * 3)],
+    account: Math.floor(Math.random() * 1000000000).toString(),
+    name: [
+      'John Doe',
+      'Jane Doe',
+      'Bob Smith',
+      'Alice Smith',
+      'John Smith',
+      'Jane Smith',
+      'Bob Doe',
+      'Alice Doe',
+    ][Math.floor(Math.random() * 8)],
+  })),
+];
+
+type Item = {
+  id: string;
+  paymentId: string;
+  amount: number;
+  status: string;
+  date: Date;
+  type: string;
+  method: string;
+  bank: string;
+  account: string;
+  name: string;
+};
+const data: TableData<Item> = {
+  nodes,
+};
+
+const TableTemplate: StoryFn<typeof TableComponent> = () => {
+  const parentRef = React.useRef<HTMLDivElement>(null);
+  return (
+    <Box
+      backgroundColor="surface.background.gray.intense"
+      padding="spacing.5"
+      minHeight="700px"
+      ref={parentRef}
+      paddingX="0"
+      paddingY="0"
+    >
+      <TableComponent data={data} isVirtualized ref={parentRef}>
+        {(tableData) => (
+          <VirtulizedWrapper
+            tableData={tableData}
+            rowHeight={(item, index) => {
+              // header height and row height
+              return index === 0 ? 50 : 57.5;
+            }}
```

Comment:
Should we have headerHeight internally defined? a bit confusing that index 0 here is header but index 0 in other places would be row item.

Since we know headerHeight, we can define it internally and leave rowHeight for consumers to define.